### DISCUSSION TITLE
Auto instrumentation for spring batch jobs and tasks

### DISCUSSION THREAD

**Discussion by ghost** [open-telemetry/opentelemetry-java-instrumentation#2663] [Q&A]
As per the release notes, auto instrumentation is supported for spring batch 3+ but we cannot see any traces related to jobs and tasklets.
Tried adding the dependency in gradle build file as below
implementation 'io.opentelemetry.javaagent.instrumentation:opentelemetry-javaagent-spring-batch-3.0:0.16.0'

Please suggest if something else needs to be done.

Thanks

**Comment by jkwatson** [comment]
Transferring this over to the instrumentation project....

**Comment by mateuszrzeszutek** [comment]
Hi @nisha-sahu,

The spring batch instrumentation is still turned off by default because of a lack of semantic conventions for batch jobs (they're still under discussion), it's treated as an "experimental" instrumentation.

You can enable it by passing `-Dotel.instrumentation.spring-batch.enabled=true` in the command line. There's no need to include any dependency in gradle.

**Reply by doppleware** [reply]
Hi, @mateuszrzeszutek! I've been trying to get a sample project to work with the OTEL agent but haven't been successful even though I've added the suggested system property.
I'm running this project: https://github.com/spring-guides/gs-batch-processing
And have added quite a lot of system properties:
```
-Dotel.javaagent.debug=true
-Dotel.instrumentation.spring-batch.enabled=true
-Dotel.instrumentation.spring-batch.item.enabled=true
-Dotel.instrumentation.spring-batch.experimental.chunk.new-trace=true
-Dotel.instrumentation.spring-batch.experimental-span-attributes=true
```

 I noticed, looking at the [code](https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/e3f90d235f1d77d4391e6f55db4002d151cae5db/instrumentation/spring/spring-batch-3.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/spring/batch/v3_0/SpringBatchInstrumentationModule.java), that unless I'm mistaken the module is always disabled?
<img width="499" alt="image" src="https://github.com/open-telemetry/opentelemetry-java-instrumentation/assets/93863/eab6ecbf-dc24-4ae4-9a55-fb1b74501054">

Would appreciate any help getting this to work.

Thanks!
ROni
